staging/src/k8s.io/apiserver/pkg/storage/etcd/etcd_watcher.go

```diff
@@ -391,7 +389,10 @@ func (w *etcdWatcher) sendModify(res *etcd.Response) {
 		// the resourceVersion to resume will never be able to get past a bad value.
 		return
 	}
-	curObjPasses := w.filter(curObj)
+	curObjPasses, err := w.pred.Matches(curObj)
```

Comment:
nit: I think it would be more readable to do:
```
curObjPasses := false
if matched, err := w.pred.Matches(curObj); err == nil && matched {
  curObjPasses = true
}
```

[also for consistency]

Reply:
@wojtek-t Addressed. PTAL. Thanks.